src/Stack/SourceMap.hs

```diff
@@ -172,7 +133,7 @@ globalsFromDump compiler = do
 
 globalsFromHints ::
        HasConfig env
-    => ActualCompiler
```

Comment:
The original type signature looks like a straight up bug, good change. Any idea why it looked like that before? (I'm well aware of the fact that I'm almost certainly the one who wrote it 😛.)

Reply:
I guess it's about this function:
```
parseActualCompiler :: T.Text -> Either PantryException ActualCompiler
parseActualCompiler = fmap wantedToActual . parseWantedCompiler
```
So `WantedCompiler` and `ActualCompiler` get parsed the same way and I'm not sure that it makes sense to have a difference here at least while we have some definitions which use parsing `ActualCompiler` from JSON (e.g. `SetupInfo`)